src/window/include/window/layer_stack.h

```diff
@@ -0,0 +1,74 @@
+#ifndef EVIE_LAYER_STACK_H_
+#define EVIE_LAYER_STACK_H_
+
+#include <iterator>
+#include <vector>
+
+#include "evie/core.h"
+#include "evie/error.h"
+#include "evie/ids.h"
+#include "evie/layer.h"
+
+namespace evie {
+
+struct LayerWrapper
+{
+  LayerID id;
+  Layer* layer;
+};
+
+/** A class that handles the layers in the engine.
+  Users can push and pop layers and overlays.
+  Overlays are still layers but sit ontop of layers in the layerstack, therefore they receive events before other
+  layers. This class will store a pointer to layers and not take ownership.
+ */
+class LayerStack
+{
+public:
+  LayerID EVIE_API PushLayer(Layer& layer);
```

Comment:
Add PushFront - PushBack - PopFront - PopBack so that you can choose where to add layers. Change to deque.